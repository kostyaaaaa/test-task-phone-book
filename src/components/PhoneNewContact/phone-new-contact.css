.phone-new-contact {
  margin-top: 20px;
}
