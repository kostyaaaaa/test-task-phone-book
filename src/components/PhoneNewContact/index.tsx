import React, { useState } from "react";
import { PhoneContactType } from "../PhoneBook";
import "./phone-new-contact.css";

type PhoneNewContactType = {
  addNewContact: (contact: PhoneContactType) => void;
};

const PhoneNewContact = ({ addNewContact }: PhoneNewContactType) => {
  const [name, setName] = useState<string>("");
  const [phone, setPhone] = useState<string>("");

  const handleNewContactName = (event: React.ChangeEvent<HTMLInputElement>) => {
    setName(event.target.value);
  };

  const handleNewContactPhone = (
    event: React.ChangeEvent<HTMLInputElement>
  ) => {
    setPhone(event.target.value);
  };

  const handleAddContact = () => {
    addNewContact({ name, phone, id: Math.random() });
    setName("");
    setPhone("");
  };

  return (
    <div className="phone-new-contact">
      <input
        className="phone-new-contact-name"
        value={name}
        placeholder="Name"
        type="text"
        onChange={handleNewContactName}
      />
      <input
        className="phone-new-contact-phone"
        value={phone}
        placeholder="Phone"
        type="text"
        onChange={handleNewContactPhone}
      />
      <button onClick={handleAddContact}>Add contact</button>
    </div>
  );
};

export default PhoneNewContact;
