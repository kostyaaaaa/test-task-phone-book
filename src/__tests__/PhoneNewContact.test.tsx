import React from "react";
import PhoneNewContact from "../components/PhoneNewContact";
import { shallow } from "enzyme";

let wrapper: any;
const addNewContact = jest.fn();

describe("PhoneNewContact", () => {
  beforeEach(() => {
    wrapper = shallow(
      <PhoneNewContact addNewContact={addNewContact}></PhoneNewContact>
    );
  });
  it("should render correctly", () => {
    expect(wrapper).toMatchSnapshot();
    expect(wrapper).not.toBeNull();
  });
  it("inputs should react on change event", () => {
    expect(wrapper.find(".phone-new-contact-name").props().value).toEqual("");
    wrapper.find(".phone-new-contact-name").simulate("change", {
      target: { value: "new user name" },
    });
    expect(wrapper.find(".phone-new-contact-name").props().value).toEqual(
      "new user name"
    );

    expect(wrapper.find(".phone-new-contact-phone").props().value).toEqual("");
    wrapper.find(".phone-new-contact-phone").simulate("change", {
      target: { value: "456456456" },
    });
    expect(wrapper.find(".phone-new-contact-phone").props().value).toEqual(
      "456456456"
    );
  });
  it("should call addNewContact function on click", () => {
    expect(addNewContact).toBeCalledTimes(0);
    wrapper.find("button").simulate("click");
    expect(addNewContact).toBeCalledTimes(1);
  });
});
