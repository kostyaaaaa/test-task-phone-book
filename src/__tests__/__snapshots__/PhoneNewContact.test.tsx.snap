// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PhoneNewContact should render correctly 1`] = `
<div
  className="phone-new-contact"
>
  <input
    className="phone-new-contact-name"
    onChange={[Function]}
    placeholder="Name"
    type="text"
    value=""
  />
  <input
    className="phone-new-contact-phone"
    onChange={[Function]}
    placeholder="Phone"
    type="text"
    value=""
  />
  <button
    onClick={[Function]}
  >
    Add contact
  </button>
</div>
`;
